### PR TITLE
Fix path_symlink only works when dir mounted to /

### LINKED ISSUE #2228: `path_symlink` doesn't work
**Describe the bug**
The `path_symlink` WASI function doesn't seem to work. Always returning `noent` if `-mount dir` doesn't explicitly specify a guest path like `-mount dir:guest`.

**To Reproduce**

Compile this Rust snippet with `wasi` crate v0.11 (for the preview1 API).

```rust
fn main() {
    unsafe {
        let base_fd = 3;

        wasi::path_symlink("file", base_fd, "new-link").unwrap();
    }
}
```

For example:

```
cargo build --target wasm32-wasi
```

And run the produced Wasm binary with Wazero, preopening a directory with a file `file` in it. But don't map it to root.

```
wazero run -mount dir a.wasm
```

**Expected behavior**
Other runtimes like Wasmtime, WasmEdge, Wasmer, and WAMR can handle such a call and create a symlink.

**Environment (please complete the relevant information):**
- Go version: 1.12.1
- wazero Version: d4a49031b671b59fb832c95741df63b96f144104
- Host architecture: amd64
- OS: Darwin

### PR DESCRIPTION
**Pull request by yagehu**
This commit fixes `path_symlink` not working when `-mount` is not supplied with a mapped directory.

fixes #2228

### CODE REVIEW

**Review by mathetake — CHANGES_REQUESTED** [review]
Please add the test case that ensures the behavior and previously failed.

File: imports/wasi_snapshot_preview1/fs_test.go [hunk 1]
```diff
@@ -4525,14 +4525,46 @@ func Test_pathSymlink_errors(t *testing.T) {
...
 
-	t.Run("success", func(t *testing.T) {
-		requireErrnoResult(t, wasip1.ErrnoSuccess, mod, wasip1.PathSymlinkName,
-			uint64(file), uint64(len(fileName)), uint64(fd), uint64(link), uint64(len(linkName)))
-		require.Contains(t, log.String(), wasip1.ErrnoName(wasip1.ErrnoSuccess))
-		st, err := os.Lstat(joinPath(dirPath, linkName))
-		require.NoError(t, err)
-		require.Equal(t, st.Mode()&os.ModeSymlink, os.ModeSymlink)
-	})
+	success_cases := []struct {
```
Comment (mathetake) [inline imports/wasi_snapshot_preview1/fs_test.go:4528]: In Go, we use CamelCase in general.

```suggestion
	successCases := []struct {
```

File: imports/wasi_snapshot_preview1/fs_test.go [hunk 2]
```diff
@@ -4525,14 +4525,46 @@ func Test_pathSymlink_errors(t *testing.T) {
...
 
-	t.Run("success", func(t *testing.T) {
-		requireErrnoResult(t, wasip1.ErrnoSuccess, mod, wasip1.PathSymlinkName,
-			uint64(file), uint64(len(fileName)), uint64(fd), uint64(link), uint64(len(linkName)))
-		require.Contains(t, log.String(), wasip1.ErrnoName(wasip1.ErrnoSuccess))
-		st, err := os.Lstat(joinPath(dirPath, linkName))
-		require.NoError(t, err)
-		require.Equal(t, st.Mode()&os.ModeSymlink, os.ModeSymlink)
-	})
+	successCases := []struct {
```
Comment (mathetake) [inline imports/wasi_snapshot_preview1/fs_test.go:4528]: this test case is named `Test_pathSymlink_errors`. Could you fix according to the change?

**Review by mathetake — APPROVED** [review]
looks good to me, waiting for others to review!

**Review by anuraaga — APPROVED** [review]
Thanks!

**Pull request merged**